### ISSUE TITLE
Interface with SPM12 Fieldmap Toolbox

### ISSUE THREAD

**Issue by MauricePasternak** [ExploreASL/ExploreASL#1625]
### Description

TL;DR ExploreASL currently doesn't apply unwarping to EPI sequences.

This feature is a sibling of #626 . The difference being that rather than interfacing with FSL's [TopUp](https://fsl.fmrib.ox.ac.uk/fsl/fslwiki/topup/TopupUsersGuide), or [Fugue/Prelude](https://fsl.fmrib.ox.ac.uk/fsl/fslwiki/FUGUE/Guide), ExploreASL sticks to its SPM roots and utilizes the [Fieldmap](https://www.fil.ion.ucl.ac.uk/spm/toolbox/fieldmap/) toolbox.

Fieldmaps are a well-known part of [BIDS](https://bids-specification.readthedocs.io/en/stable/modality-specific-files/magnetic-resonance-imaging-data.html#expressing-the-mr-protocol-intent-for-fieldmaps), coming in 4 main flavors:
1) A precomputed phase-difference image + 1 or 2 magnitude images (commonly found in Siemens)
2) Two phase maps and two magnitude images (commonly found in Philips)
3) A precomputed fieldmap (commonly found in GE)
4) 2+ EPI scans with different phase encoding

^ Each of these likely deserves 1 or more sub-issues.

### Tasks

- [ ] Implement functionality for grabbing any additional information off DICOM headers that `dcm2niix` fails (i.e. acceleration factors listed in Phoenix Protocol), especially parameters that assist in making an "educated guess" about blip direction
- [ ] Implement functionality for detecting fieldmaps in rawdata
- [ ] Implement functionality for determining fieldmap flavor
- [ ] Implement functionality for translating to "legacy" format
- [ ] Implement functionality for fieldmap creation in the case of flavors 1,2, or 4
- [ ] Implement functionality for applying fieldmaps to ASL4D prior to realignment/reslicing

### Additional Notes

- SPM appears to have a different definition of `TotalReadoutTime` versus FSL & BIDS. SPM's definition looks to be roughly: 

EchoSpacing * AcquisitionMatrix_PhaseDirectionVal / ParallelReductionFactorInPlane

The above applies to Siemens. The denominator is sometimes referred to under different names such as GRAPPA. Again, if `dcm2niix` fails to grab some of these, it may be prudent to double-check.

- If the manufacturer is detected as Siemens and we are working with fieldmap flavor (1) it may be worthwhile to assume a difference in echo time of 2.46ms as that is the most common setting.
- Here is a [good thread](https://neurostars.org/t/phaseencodingdirection-in-json-file/20259/2) to assist with best-guessing blip direction based on acquisition

### HENK: this actually includes two issues:
1. If we should move from FSL TopUp to SPM fieldmap (or Hysco/similar SPM-based unwarping toolboxes)
   Note that — unlike what Maurice writes above — we currently use FSL's TopUp, not any of the other options.
2. To implement non-PePolar unwrapping options as well
